ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -115,16 +98,160 @@ object ParallelIndexerFactory {
             ).map(_ => ())
 
       def subscribe(readService: ReadService): Future[Source[Unit, NotUsed]] =
-        postgresDaoPool
-          .execute(_.initialize)
-          .map { case (offset, eventSeqId) =>
-            ingest(eventSeqId.getOrElse(0L))(readService.stateUpdates(beginAfter = offset))
-          }(scala.concurrent.ExecutionContext.global)
+        dbDispatcher
+          .executeSql(metrics.daml.parallelIndexer.initialization)(
+            storageBackend.initialize
+          ) // TODO here is an implicit LoggingContext injected from the factory apply implicit param, is that good this way? shouldn't we create here something for the purpose?
```

Comment:
I don't understand this TODO. Do you see an issue with the logging context from `ParallelIndexerFactory.apply` being passed to `dbDispatcher.executeSql`?

Reply:
well, not, I am happy to remove this entirely if there is none. I was just concerned because executeSql needs this implicit param, so in theory we could do something like setting logging params in the context regarding whatnot before we hit the ingestion